arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitServiceImpl.java

```diff
@@ -132,54 +141,47 @@ public void releaseCasesOfContext(ReplayPlan replayPlan,
   }
 
   private void doSendValuesToRemoteHost(List<ReplayActionCaseItem> values,
-      ExecutionStatus executionStatus) {
+                                        PlanExecutionContext<?> planExecutionContext) {
+
+    List<ServiceInstance> targetServiceInstances = planExecutionContext.getCurrentTargetInstances();
+    if (CollectionUtils.isEmpty(targetServiceInstances)) {
+      LOGGER.warn("The target instance list is empty,skip send.");
+      markAllSendStatus(values, CaseSendStatusType.READY_DEPENDENCY_FAILED);
+      return;
+    }
+
     final int valueSize = values.size();
-    final SendSemaphoreLimiter semaphore = executionStatus.getLimiter();
     final CountDownLatch groupSentLatch = new CountDownLatch(valueSize);
+    ArrayBlockingQueue<ReplayActionCaseItem> arrayBlockingQueue = new ArrayBlockingQueue<>(valueSize, false, values);
 
-    for (int i = 0; i < valueSize; i++) {
-      ReplayActionCaseItem replayActionCaseItem = values.get(i);
-      ReplayActionItem actionItem = replayActionCaseItem.getParent();
-      MDCTracer.addDetailId(replayActionCaseItem.getId());
-      // checkpoint: before init case runnable
-      if (executionStatus.isAbnormal()) {
-        LOGGER.info("replay interrupted,case item id:{}", replayActionCaseItem.getId());
-        MDCTracer.removeDetailId();
-        return;
-      }
+    // calculate the number of distribution tasks
+    int distributeTaskNum = Math.min(valueSize, targetServiceInstances.size());
```

Comment:
`groupSentLatch` count is valueSize, but `distributeTaskNum` maybe targetServiceInstances.size,  
if different, groupSentLatch.await maybe never **end**？
the number of times a method `doDistribute` is called in a for loop and `groupSentLatch` count init value may be inconsistent.